feat(components): added common components along with demo for ease of use.

JVP555 opened the pull request:
feat(components): added common components along with demo for ease of use.

component use (component_import_name)

Data-table which have filter by column for search and by date, import_name (DataTable)
Date with input and calender , import_name (DatePicker)
Phone Number with country code and number input , import_name (PhoneNumber)
variable Select list scrollable , import_name (SelectList)
variable dropdown , import_name (Dropdown)
Placeholder , import_name (Placeholder)


current look of a page :
<img width="1919" height="1049" alt="image" src="https://github.com/user-attachments/assets/5c11a276-10d3-421e-966b-617a8076fc75" />

alpha-og commented:
Demo of the components need not be included explicitly and must be exhibited as part of feature completions/ introductions. Providing adequate images/ videos of all changes made are sufficient.

alpha-og commented:
@RoshinR2005 @RohanAJacob2004 What are your thoughts on the need for a reference spec implementation for frontend components? 

I understand the utility but I think this will go stale very quickly. Developers can use already completed components for reference, eliminating the benefit of a dedicated reference spec implementation (not that we have a reference spec for the UI). It is the sole responsibility of the developer to understand the utility of any and all components they use and implement them without producing regressions.

JVP555 commented:
I have updated the routes and naming as per the latest specifications and moved the demo sections into a separate branch (reference/components). This way, they remain available as optional reference material without impacting the main codebase.
